apps/src/lib/levelbuilder/ai-tutor/AITutorTester.tsx

```diff
@@ -63,12 +69,46 @@ const AITutorTester: React.FC<AITutorTesterProps> = ({allowed}) => {
     for (let i = 0; i < data.length; i++) {
       if (selectedEndpoint === 'llm-guard') {
         getLLMGuardToxicity(data[i]);
+      } else if (genAIEndpointIds.includes(selectedEndpoint)) {
```

Comment:
If you use a concurrency method like `Promise.allSettled`, you can do something like:

```
const getAIResponses = async () => {
  setResponsesPending(true);

  const tasks = data.map(async (item) => {
    if (selectedEndpoint === 'llm-guard') {
      return getLLMGuardToxicity(item);
    } else if (genAIEndpointIds.includes(selectedEndpoint)) {
      return getGenAIResponses(item);
    } else {
      return askAITutor(item);
    }
  });

  const results = await Promise.allSettled(tasks);
  setResponsesPending(false); 
};
```
I think it makes how you're handling the asynchronicity a little more explicit 🤷‍♀️

It also allows you to easily filter for successes and failures, if desired.
```
// Count successful responses
const successfulResponses = results.filter(result => result.status === 'fulfilled').length;
``` 

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled